editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -2097,8 +2098,14 @@ bool CanvasItemEditor::_gui_input_move(const Ref<InputEvent> &p_event) {
 		}
 
 		// Confirm the move (only if it was moved)
-		if (b.is_valid() && !b->is_pressed() && b->get_button_index() == MouseButton::LEFT) {
-			if (transform.affine_inverse().xform(b->get_position()) != drag_from) {
+		if (tool_changed || (b.is_valid() && !b->is_pressed() && b->get_button_index() == MouseButton::LEFT)) {
+			bool was_dragged = false;
+			if (b.is_valid() && !b->is_pressed() && b->get_button_index() == MouseButton::LEFT) {
```

Comment:
```suggestion
			if (!tool_changed) {
```
Though I'd swap it around, no need to check again

Reply:
Good point